backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala

```diff
@@ -223,48 +223,32 @@ object BackendSettings extends BackendSettingsApi {
   }
 
   /**
-   * Check whether plan is Count(1).
+   * Check whether a plan needs to be offloaded even though they have empty input schema, e.g,
+   * Sum(1), Count(1), rand(), etc.
    * @param plan:
    *   The Spark plan to check.
-   * @return
-   *   Whether plan is an Aggregation of Count(1).
    */
-  private def isCount1(plan: SparkPlan): Boolean = {
+  private def mayNeedOffload(plan: SparkPlan): Boolean = {
     plan match {
-      case exec: HashAggregateExec
-          if exec.aggregateExpressions.nonEmpty &&
-            exec.aggregateExpressions.forall(
-              expression => {
-                expression.aggregateFunction match {
-                  case c: Count => c.children.size == 1 && c.children.head.equals(Literal(1))
-                  case _ => false
-                }
-              }) =>
-        true
-      case _ =>
-        false
-    }
-  }
-
-  /**
-   * Check whether plan is Sum(1).
-   * @param plan:
-   *   The Spark plan to check.
-   * @return
-   *   Whether plan is an Aggregation of Sum(1).
-   */
-  private def isSum1(plan: SparkPlan): Boolean = {
-    plan match {
-      case exec: HashAggregateExec
-          if exec.aggregateExpressions.nonEmpty &&
-            exec.aggregateExpressions.forall(
-              expression => {
-                expression.aggregateFunction match {
-                  case s: Sum => s.children.size == 1 && s.children.head.equals(Literal(1))
-                  case _ => false
-                }
-              }) =>
-        true
+      case exec: HashAggregateExec if exec.aggregateExpressions.nonEmpty =>
+        // Check Sum(1) or Count(1).
+        exec.aggregateExpressions.forall(
+          expression => {
+            val aggFunction = expression.aggregateFunction
+            aggFunction match {
+              case _: Sum | _: Count =>
+                aggFunction.children.size == 1 && aggFunction.children.head.equals(Literal(1))
+              case _ => false
+            }
+          })
+      case p: ProjectExec if p.projectList.nonEmpty =>
+        p.projectList.forall(
+          expr =>
+            expr match {
+              // Block directly falling back the below functions by FallbackEmptySchemaRelation.
+              case _: Rand => true
+              case _ => false
+            })
```

Comment:
these changes seems not related with `rand()` ?

Reply:
As `rand()` has no child output, we need this to avoid make it fall back by `fallbackOnEmptySchema` rule.